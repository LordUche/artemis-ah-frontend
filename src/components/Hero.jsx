import React, { Fragment, Component } from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import {
  object, shape, func, bool, string, number
} from 'prop-types';
import logo from '../assets/img/logo.svg';
import illustration from '../assets/img/illustration.svg';
import NavDropdown from './NavDropdown';
import AHLoginModal from '../views/LoginModal';
import Hamburger from './Hamburger';
import Button from './Button';
/**
 * @description Hero - Hero component for Landing Page view
 * @param {object} e event object
 * @returns {JSX} - JSX component
 */
class Hero extends Component {
  state = {
    displaySearchBar: false,
    searchQuery: '',
    showResponsiveNav: false
  };

  toggleResponsiveNav = () => {
    const { showResponsiveNav } = this.state;
    this.setState({
      showResponsiveNav: !showResponsiveNav
    });
  };

  searchButtonOnclick = () => {
    this.setState({
      displaySearchBar: true
    });
  };

  hideSearchField = (e) => {
    if (
      e.target.name !== 'searchQuery'
      && e.target.id !== 'search'
      && (e.target.id !== 'searchIconChild' && e.target.id !== 'searchIcon')
    ) {
      this.setState({ displaySearchBar: false });
    }
  };

  searchKeyword = () => {
    const { searchQuery } = this.state;

    if (searchQuery) {
      this.setState({ searchQuery: 'Redirects to the search page' });
    }
  };

  handleKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.searchKeyword();
    }
  };

  handleSearchBarChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  /**
   * @description Hero - Hero component for Landing Page view
   * @returns {JSX} - JSX component
   */
  render() {
    const { displaySearchBar, showResponsiveNav, searchQuery } = this.state;
    const {
      smoothScrollListener,
      showLoginModal,
      revealLoginModal,
      hideLoginModal,
      isLoggedIn,
      history,
      toggleSignUpModal
    } = this.props;

    return (
      <section className="hero" onClick={this.hideSearchField} role="presentation">
        {showLoginModal && (
          <AHLoginModal onClose={hideLoginModal} toggleSignUpModal={toggleSignUpModal} />
        )}
        <div className="hero__description">
          <div className="hero__header">
            <Router>
              <Link to="./" className="hero__logo">
                <img src={logo} alt="Authors Haven Logo" />
              </Link>
            </Router>
            <Hamburger open={showResponsiveNav} toggleMenu={this.toggleResponsiveNav}>
              <li>
                <span
                  id="login-link"
                  className="link_lookalike"
                  role="presentation"
                  onClick={revealLoginModal}
                >
                  Login
                </span>
              </li>
              <li>
                <span
                  id="signup-link"
                  className="link_lookalike"
                  role="presentation"
                  onClick={toggleSignUpModal}
                >
                  Register
                </span>
              </li>
              <li>
                <Router>
                  <Link to="./register">Explore</Link>
                </Router>
              </li>
            </Hamburger>
          </div>
          <div className="hero__text">
            <h2 className="hero__text--large">Authors Haven</h2>
            <p className="hero__text--normal">Building a community of like minded authors.</p>
          </div>
          <div className="hero__cta">
            <Button
              btnText="Explore Stories"
              customClass="btn btn-explore"
              btnType="button"
              onClick={() => history.push('./explore')}
            />

            <Button
              btnText="Start Writing"
              customClass="btn btn-write"
              btnType="button"
              onClick={isLoggedIn ? () => history.push('./create-article') : revealLoginModal}
            />
          </div>
        </div>
        <div className="hero__illustration">
          <nav className="hero__nav">
            {!displaySearchBar ? (
              <ul className="hero__nav--links">
                {!isLoggedIn && (
                  <Fragment>
                    <li>
                      <span
                        id="login-link"
                        className="link_lookalike"
                        role="presentation"
                        onClick={revealLoginModal}
                      >
                        Login
                      </span>
                    </li>
                    <li>
                      <span
                        id="signup-link"
                        className="link_lookalike"
                        role="presentation"
                        onClick={toggleSignUpModal}
                      >
                        Register
                      </span>
                    </li>
                  </Fragment>
                )}
                <NavDropdown parentLinkName="Explore">
                  <li>
                    <div
                      role="button"
                      tabIndex={0}
                      onKeyPress={() => history.push('./explore/Food')}
                      onClick={() => history.push('./explore/Food')}
                    >
                      <span className="link_lookalike">Food</span>
                    </div>
                  </li>
                  <li>
                    <div
                      role="button"
                      tabIndex={0}
                      onKeyPress={() => history.push('./explore/Technology')}
                      onClick={() => history.push('./explore/Technology')}
                    >
                      <span className="link_lookalike">Technology</span>
                    </div>
                  </li>
                  <li>
                    <div
                      role="button"
                      tabIndex={0}
                      onKeyPress={() => history.push('./explore/Health')}
                      onClick={() => history.push('./explore/Technology')}
                    >
                      <span className="link_lookalike">Health</span>
                    </div>
                  </li>
                  <li>
                    <div
                      role="button"
                      tabIndex={0}
                      onKeyPress={() => history.push('./explore/Finance')}
                      onClick={() => history.push('./explore/Finance')}
                    >
                      <span className="link_lookalike">Finance</span>
                    </div>
                  </li>
                  <li>
                    <div
                      role="button"
                      tabIndex={0}
                      onKeyPress={() => history.push('./explore/Arts')}
                      onClick={() => history.push('./explore/Arts')}
                    >
                      <span className="link_lookalike">Arts</span>
                    </div>
                  </li>
                </NavDropdown>
                <li>
                  <span role="presentation" onClick={this.searchButtonOnclick} id="searchIcon">
                    <i className="fas fa-search hero__nav--search" id="searchIconChild" />
                  </span>
                </li>
              </ul>
            ) : (
              <span className="hero__nav--search_wrapper" role="presentation">
                <input
                  className="hero__nav--search_input"
                  placeholder="Type a keyword..."
                  type="text"
                  onChange={this.handleSearchBarChange}
                  onKeyPress={this.handleKeyPress}
                  name="searchQuery"
                  value={searchQuery}
                />
                <i
                  className="fas fa-search hero__nav--search"
                  id="search"
                  onClick={this.searchKeyword}
                  role="presentation"
                />
              </span>
            )}
            <img src={illustration} alt="Illustration of a reader" />
          </nav>
        </div>
        <button type="button" className="scroll" onClick={smoothScrollListener}>
          <i className="fas fa-angle-down" />
        </button>
      </section>
    );
  }
}

Hero.propTypes = {
  smoothScrollListener: func.isRequired,
  showLoginModal: bool.isRequired,
  revealLoginModal: func.isRequired,
  hideLoginModal: func.isRequired,
  isLoggedIn: bool.isRequired,
  toggleSignUpModal: func.isRequired,
  history: shape({
    action: string,
    block: func,
    createHref: func,
    go: func,
    goBack: func,
    goForward: func,
    length: number,
    listen: func,
    location: object,
    push: func,
    replace: func
  }).isRequired
};

export default Hero;
