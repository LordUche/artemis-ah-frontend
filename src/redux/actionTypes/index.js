export const CHANGE_NAME = 'CHANGE_NAME';

// Get all articles Action types
export const GET_ARTICLES = 'GET_ARTICLES';
export const GET_ARTICLES_ERROR = 'GET_ARTICLES_ERROR';
export const ARTICLES_LOADING = 'ARTICLES_LOADING';

// Auth Action types
export const LOGIN_USER = 'LOGIN_USER';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const CLEAR_AUTH_ERROR = 'CLEAR_AUTH_ERROR';
export const AUTH_LOADING = 'AUTH_LOADING';

// Profile action types
export const PROFILE_USER_DETAILS_FETCHED = 'PROFILE_USER_DETAILS_FETCHED';
export const PROFILE_USER_DETAILS_FETCH_ERROR = 'PROFILE_USER_DETAILS_FETCH_ERROR';

export const PROFILE_ARTICLES_FETCHING = 'PROFILE_ARTICLES_FETCHING';
export const PROFILE_ARTICLES_FETCHED = 'PROFILE_ARTICLES_FETCHED';
export const PROFILE_ARTICLES_FETCH_ERROR = 'PROFILE_ARTICLES_FETCH_ERROR';

export const PROFILE_FOLLOWERS_FETCHING = 'PROFILE_FOLLOWERS_FETCHING';
export const PROFILE_FOLLOWERS_FETCHED = 'PROFILE_FOLLOWERS_FETCHED';
export const PROFILE_FOLLOWERS_FETCH_ERROR = 'PROFILE_FOLLOWERS_FETCH_ERROR';

export const PROFILE_FOLLOWING_FETCHING = 'PROFILE_FOLLOWING_FETCHING';
export const PROFILE_FOLLOWING_FETCHED = 'PROFILE_FOLLOWING_FETCHED';
export const PROFILE_FOLLOWING_FETCH_ERROR = 'PROFILE_FOLLOWING_FETCH_ERROR';

export const PROFILE_DETAILS_UPDATING = 'PROFILE_DETAILS_UPDATING';
export const PROFILE_DETAILS_UPDATED = 'PROFILE_DETAILS_UPDATED';
export const PROFILE_DETAILS_UPDATE_ERROR = 'PROFILE_DETAILS_UPDATE_ERROR';

export const PROFILE_RESET_EDIT_STATE = 'PROFILE_RESET_EDIT_STATE';

export const PROFILE_RESET = 'PROFILE_RESET';
// Articles Action types
export const FETCH_TAGS = 'FETCH_TAGS';
export const FETCH_TAGS_ERROR = 'FETCH_TAGS_ERROR';
export const CREATE_ARTICLE = 'CREATE_ARTICLE';
export const CREATE_ARTICLE_ERROR = 'CREATE_ARTICLE_ERROR';
export const CLEAR_ARTICLE_ERROR = 'CLEAR_ARTICLE_ERROR';
export const PUBLISHING_ARTICLE = 'PUBLISHING_ARTICLE';

// Sign Up Action types
export const SIGN_UP = 'SIGN_UP';
export const SIGN_UP_ERROR = 'SIGN_UP_ERROR';

export const EDIT_ARTICLE = 'EDIT_ARTICLE';
export const SAVE_EDITED_ARTICLE = 'SAVE_EDITED_ARTICLE';

export const OPEN_DELETE_CONFIRMATION_MODAL = 'OPEN_DELETE_CONFIRMATION_MODAL';
export const CLOSE_DELETE_CONFIRMATION_MODAL = 'CLOSE_DELETE_CONFIRMATION_MODAL';
export const DELETE_ARTICLE = 'DELETE_ARTICLE';

export const FETCH_DELETE_ERROR = 'FETCH_DELETE_ERROR';

// Record Action types
export const GETTING_ARTICLE = 'GETTINg_ARTICLE';
export const GOT_ARTICLE = 'GOT_ARTICLE';
export const ERROR_GETTING_ARTICLE = 'ERROR_GETTING_ARTICLE';

export const NEW_NOTIFICATION = 'NEW_NOTIFICATION';
export const GET_NOTIFICATIONS = 'GET_NOTIFICATIONS';
export const READ_NOTIFICATION = 'READ_NOTIFICATION';
