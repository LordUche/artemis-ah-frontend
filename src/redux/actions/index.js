// Base url
const BASE_URL = 'https://authorshaven.herokuapp.com/api';

export default BASE_URL;
