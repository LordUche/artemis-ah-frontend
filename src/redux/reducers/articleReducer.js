import {
  CREATE_ARTICLE,
  CREATE_ARTICLE_ERROR,
  CLEAR_ARTICLE_ERROR,
  PUBLISHING_ARTICLE,
  SAVE_EDITED_ARTICLE,
  EDIT_ARTICLE,
  OPEN_DELETE_CONFIRMATION_MODAL,
  CLOSE_DELETE_CONFIRMATION_MODAL,
  GET_ARTICLES,
  GET_ARTICLES_ERROR,
  GETTING_ARTICLE,
  GOT_ARTICLE,
  ERROR_GETTING_ARTICLE
} from '../actionTypes';

export const initialState = {
  articleData: {},
  articles: [],
  totalNumberOfArticles: 0,
  limit: 0,
  loading: true,
  errors: {},
  isPublishing: false,
  articleCardData: JSON.parse(localStorage.getItem('cardData')) || {},
  confirmationModal: false,
  updatedArticle: {},
  isGetting: false,
  articleGotten: {},
  newArticleSlug: ''
};

/**
 * @param {object} state - The initial state
 * @param {object} {*} - destructured type object
 * @returns {object} - The transformed state
 */
const articleReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case GET_ARTICLES:
      return {
        ...state,
        articles: payload.articles,
        totalNumberOfArticles: payload.total,
        limit: payload.limit,
        loading: false
      };
    case GET_ARTICLES_ERROR:
      return {
        ...state,
        errors: payload,
        loading: false
      };
    case CREATE_ARTICLE:
      return {
        ...state,
        isPublishing: false,
        newArticleSlug: payload.article.slug,
        articleData: payload.article
      };
    case CREATE_ARTICLE_ERROR:
      return {
        ...state,
        isPublishing: false,
        errors: payload
      };
    case CLEAR_ARTICLE_ERROR:
      return {
        ...state,
        isPublishing: false,
        isGetting: false,
        newArticleSlug: '',
        errors: {}
      };
    case PUBLISHING_ARTICLE:
      return {
        ...state,
        isPublishing: true
      };
    case EDIT_ARTICLE:
      return {
        ...state,
        articleCardData: payload
      };
    case SAVE_EDITED_ARTICLE:
      return {
        ...state,
        isPublishing: false,
        updatedArticle: payload
      };
    case OPEN_DELETE_CONFIRMATION_MODAL:
      return {
        ...state,
        confirmationModal: true
      };
    case CLOSE_DELETE_CONFIRMATION_MODAL:
      return {
        ...state,
        confirmationModal: false
      };
    case GETTING_ARTICLE:
      return {
        ...state,
        isGetting: true,
        errors: {}
      };
    // eslint-disable-next-line no-case-declarations
    case GOT_ARTICLE:
      const { article, clap } = payload;
      return {
        ...state,
        isGetting: false,
        articleGotten: { ...article, clap },
        errors: {}
      };
    case ERROR_GETTING_ARTICLE:
      return {
        ...state,
        isGetting: false,
        articleDetailsGotten: {},
        errors: payload
      };
    default:
      return state;
  }
};

export default articleReducer;
