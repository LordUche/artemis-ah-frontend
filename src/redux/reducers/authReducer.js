import {
  LOGIN_USER, LOGIN_ERROR, AUTH_LOADING, CLEAR_AUTH_ERROR
} from '../actionTypes';

/**
 * @description function to return initial state
 * @param {object} localStorage the local storage to query (to be mocked for testing)
 * @returns {object} the initial state
 */
export const initialState = localStorage => ({
  isLoggedIn: !!localStorage.authorsHavenToken,
  token: localStorage.authorsHavenToken || '',
  errorMessages: {},
  loading: false
});

/**
 * @description reducer for auth operations
 * @param {object} state the current state
 * @param {object} action the action dispatched
 * @returns {object} the new state
 */
const authReducer = (state = initialState(localStorage), { type, payload }) => {
  switch (type) {
    case LOGIN_USER:
      return {
        ...state,
        token: payload.token,
        isLoggedIn: true,
        loading: false
      };
    case AUTH_LOADING:
      return {
        ...state,
        loading: true
      };
    case LOGIN_ERROR:
      return {
        ...state,
        errorMessages: payload.message ? { errors: { general: payload.message } } : payload,
        loading: false
      };
    case CLEAR_AUTH_ERROR:
      return {
        ...state,
        errorMessages: []
      };
    default:
      return state;
  }
};

export default authReducer;
