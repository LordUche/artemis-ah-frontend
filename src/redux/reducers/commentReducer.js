import {
  GET_COMMENTS,
  POST_COMMENT,
  POST_COMMENT_ERROR,
  COMMENT_LOADING,
  CLEAR_POSTED,
  LIKE_COMMENT_ERROR,
  TOGGLE_COMMENT_LIKE,
} from '../actionTypes';


export const initialState = {
  articleComments: [],
  errors: {},
  posted: false,
  loading: false
};

/**
 * @param {*} state
 * @returns {object} comment reducer
 */
export const commentReducer = (state = initialState, { type, payload }) => {
  let chosenComment, remainingComments;
  if (type === TOGGLE_COMMENT_LIKE || type === LIKE_COMMENT_ERROR) {
    const { articleComments } = state;
    const { id } = payload;
    chosenComment = articleComments.find(c => c.id === id);
    remainingComments = articleComments.filter(c => c.id !== id);
  }
  switch (type) {
    case GET_COMMENTS:
      return {
        ...state,
        articleComments: payload,
        errors: {}
      };
    case POST_COMMENT:
      return {
        ...state,
        posted: true,
        loading: false,
        errors: {}
      };
    case POST_COMMENT_ERROR:
      return {
        ...state,
        errors: payload.errors,
        loading: false
      };
    case COMMENT_LOADING:
      return {
        ...state,
        loading: true,
        errors: {}
      };
    case CLEAR_POSTED:
      return {
        ...state,
        posted: false,
        errors: {}
      };
    case TOGGLE_COMMENT_LIKE:
      chosenComment.hasLiked = !chosenComment.hasLiked;
      chosenComment.totalLikes = chosenComment.hasLiked
        ? chosenComment.totalLikes + 1 : chosenComment.totalLikes - 1;
      return {
        ...state,
        articleComments: [...remainingComments, chosenComment].sort((a, b) => b.id - a.id)
      };
    case LIKE_COMMENT_ERROR:
      chosenComment.hasLiked = !chosenComment.hasLiked;
      chosenComment.totalLikes = chosenComment.hasLiked
        ? chosenComment.totalLikes + 1 : chosenComment.totalLikes - 1;
      return {
        ...state,
        articleComments: [...remainingComments, chosenComment].sort((a, b) => b.id - a.id),
        errors: { message: 'Cannot perform action right now, please check your connection and try again later' }
      };
    default:
      return state;
  }
};

export default commentReducer;
