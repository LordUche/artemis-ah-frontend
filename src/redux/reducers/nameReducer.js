import CHANGE_NAME from '../actionTypes/index';

const inttialState = {
  name: 'adaeze'
};

/**
 * @description change name reducer
 * @param {*} state
 * @param {*} action
 * @returns {object} state
 */
export default function changeName(state = inttialState, action) {
  switch (action.type) {
    case CHANGE_NAME:
      return { ...state, name: action.payload };
    default:
      return state;
  }
}
