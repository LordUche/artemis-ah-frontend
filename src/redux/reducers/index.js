import { combineReducers } from 'redux';
import authReducer from './authReducer';
import userReducer from './userReducer';
import profileReducer from './profileReducer';
import tagReducer from './tagReducer';
import articleReducer from './articleReducer';
import notificationsReducer from './notificationsReducer';
import commentsReducer from './commentReducer';

const rootReducer = combineReducers({
  auth: authReducer,
  user: userReducer,
  profile: profileReducer,
  tags: tagReducer,
  article: articleReducer,
  notifications: notificationsReducer,
  comments: commentsReducer,
});

export default rootReducer;
