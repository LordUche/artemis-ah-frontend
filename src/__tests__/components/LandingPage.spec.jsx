import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { createStore, applyMiddleware } from 'redux';
import ReduxPromise from 'redux-promise';
import sinon from 'sinon';
import LandingPageStore, { LandingPage } from '../../views/LandingPage';
import reducers from '../../redux/reducers';
import Hero from '../../components/Hero';

const { smoothScrollToAbout } = new LandingPage();

describe('Landing Page Component', () => {
  const store = createStore(reducers, applyMiddleware(ReduxPromise));
  it('should have match the given snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <LandingPageStore />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should test for class methods', () => {
    // const onClickMock = jest.fn();
    const mockRevealModal = jest.fn();
    const mockHideModal = jest.fn();
    const mockAction = jest.fn();
    const landing = shallow(<LandingPage loginUserViaSocialMedia={mockAction} />);
    const hero = shallow(
      <Hero
        smoothScrollListener={() => smoothScrollToAbout()}
        showLoginModal={false}
        revealLoginModal={mockRevealModal}
        hideLoginModal={mockHideModal}
        isLoggedIn={false}
      />
    );

    sinon.stub(window, 'scrollBy');
    landing.instance().hideLoginModal();
    landing.instance().revealLoginModal();
    landing.instance().componentWillMount();
    landing.instance().validateURLPayload();
    expect(landing.find('Hero').exists()).toBe(true);
    expect(hero.find('.scroll').exists()).toBe(true);
    hero.find('.scroll').simulate('click');
  });
});
