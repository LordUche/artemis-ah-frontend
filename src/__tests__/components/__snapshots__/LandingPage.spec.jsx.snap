// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Component should have match the given snapshot 1`] = `
Array [
  <section
    className="hero"
    onClick={[Function]}
    role="presentation"
  >
    <div
      className="hero__description"
    >
      <div
        className="hero__header"
      >
        <a
          className="hero__logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Authors Haven Logo"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </a>
        <div
          className="hamburger_div"
        >
          <i
            className="fas fa-bars"
            id="hamburger"
            onClick={[Function]}
            role="presentation"
          />
          <section
            className="hamburger_div_menu"
          >
            <aside
              className="hamburger_div_menu_aside hamburger_div_menu_aside_hide"
            >
              <span
                id="hamburger-ex"
                onClick={[Function]}
                role="presentation"
              >
                ×
              </span>
              <ul
                onClick={[Function]}
                role="presentation"
              >
                <li>
                  <span
                    className="link_lookalike"
                    id="login-link"
                    onClick={[Function]}
                    role="presentation"
                  >
                    Login
                  </span>
                </li>
                <li>
                  <a
                    href="/register"
                    onClick={[Function]}
                  >
                    Register
                  </a>
                </li>
                <li>
                  <a
                    href="/register"
                    onClick={[Function]}
                  >
                    Explore
                  </a>
                </li>
              </ul>
            </aside>
          </section>
        </div>
      </div>
      <div
        className="hero__text"
      >
        <h2
          className="hero__text--large"
        >
          Authors Haven
        </h2>
        <p
          className="hero__text--normal"
        >
          Building a community of like minded authors.
        </p>
      </div>
      <div
        className="hero__cta"
      >
        <a
          className="btn"
          href="/explore"
          onClick={[Function]}
        >
          Explore Stories
        </a>
        <a
          className="btn"
          href="/create-story"
          onClick={[Function]}
        >
          Start Writing
        </a>
      </div>
    </div>
    <div
      className="hero__illustration"
    >
      <nav
        className="hero__nav"
      >
        <ul
          className="hero__nav--links"
        >
          <li>
            <span
              className="link_lookalike"
              id="login-link"
              onClick={[Function]}
              role="presentation"
            >
              Login
            </span>
          </li>
          <li>
            <a
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
          <li
            className="dropdown__trigger"
          >
            <a
              href="/explore"
              id="explore"
              onClick={[Function]}
              onMouseLeave={[Function]}
            >
              Explore
               
              <i
                className="fas fa-angle-down"
              />
            </a>
            <ul
              className="dropdown__body "
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <li>
                <a
                  href="/filter?tag=Food"
                  onClick={[Function]}
                >
                  Food
                </a>
              </li>
              <li>
                <a
                  href="/filter?tag=Technology"
                  onClick={[Function]}
                >
                  Technology
                </a>
              </li>
              <li>
                <a
                  href="/filter?tag=Health"
                  onClick={[Function]}
                >
                  Health
                </a>
              </li>
              <li>
                <a
                  href="/filter?tag=Finance"
                  onClick={[Function]}
                >
                  Finance
                </a>
              </li>
              <li>
                <a
                  href="/filter?tag=Arts"
                  onClick={[Function]}
                >
                  Arts
                </a>
              </li>
            </ul>
          </li>
          <li>
            <span
              id="searchIcon"
              onClick={[Function]}
              role="presentation"
            >
              <i
                className="fas fa-search hero__nav--search"
                id="searchIconChild"
              />
            </span>
          </li>
        </ul>
        <img
          alt="Illustration of a reader"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </nav>
    </div>
    <button
      className="scroll"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-angle-down"
      />
    </button>
  </section>,
  <section
    className="segment about"
  >
    <h3
      className="segment__title"
    >
      How Authors Haven works
    </h3>
    <p
      className="segment__text"
    >
      Create and share your story with an amazing community of authors and avid readers.
    </p>
    <div
      className="steps"
    >
      <div
        className="step"
      >
        <span
          className="step__number"
        >
          1
        </span>
        <span
          className="step__title"
        >
          Write
        </span>
        <span
          className="step__text"
        >
          Share your unique story with readers.
        </span>
      </div>
      <div
        className="step"
      >
        <span
          className="step__number"
        >
          2
        </span>
        <span
          className="step__title"
        >
          Explore
        </span>
        <span
          className="step__text"
        >
          Find stories from amazing authors.
        </span>
      </div>
      <div
        className="step"
      >
        <span
          className="step__number"
        >
          3
        </span>
        <span
          className="step__title"
        >
          Connect
        </span>
        <span
          className="step__text"
        >
          Engage with a community of writers.
        </span>
      </div>
    </div>
  </section>,
  <section
    className="segment featured"
  >
    <h3
      className="segment__title"
    >
      Featured Categories
    </h3>
    <p
      className="segment__text"
    >
      Create and share your story with an amazing community of authors and avid readers.
    </p>
    <div
      className="featured__articles"
    >
      <div
        className="featured__card"
      >
        <span
          className="category"
        >
          Food
        </span>
        <span
          className="title"
        >
          Top recipes to making amazing sandwiches.
        </span>
      </div>
      <div
        className="featured__card"
      >
        <span
          className="category"
        >
          Technology
        </span>
        <span
          className="title"
        >
          Top recipes to making amazing sandwiches.
        </span>
      </div>
      <div
        className="featured__card"
      >
        <span
          className="category"
        >
          Arts
        </span>
        <span
          className="title"
        >
          Top recipes to making amazing sandwiches.
        </span>
      </div>
      <div
        className="featured__card"
      >
        <span
          className="category"
        >
          Health
        </span>
        <span
          className="title"
        >
          Top recipes to making amazing sandwiches.
        </span>
      </div>
      <div
        className="featured__card"
      >
        <span
          className="category"
        >
          Finance
        </span>
        <span
          className="title"
        >
          Top recipes to making amazing sandwiches.
        </span>
      </div>
    </div>
    <a
      className="btn"
      href="/explore"
      onClick={[Function]}
    >
      Explore Stories
       
      <i
        className="fas fa-angle-right"
      />
    </a>
  </section>,
  <footer>
    ©
     
    2019
     
    Authors Haven
  </footer>,
]
`;
